### PR TITLE
Feature/aa 527 api for supplier accommodation separation

### PR DESCRIPTION
**Pull request by hasmikharutyunyan**

### CODE REVIEW

**Review by AlexeyShok — APPROVED** [review]

File: HappyTravel.Nakijin.Api/Controllers/AccommodationManagementController.cs [hunk 1]
```diff
@@ -56,8 +56,8 @@ public AccommodationsManagementController(IAccommodationManagementService accomm
         /// <param name="accommodation"></param>
         /// <returns></returns>
         [HttpPost("accommodations/{accommodationId}/manual-correction")]
-        [ProducesResponseType((int) HttpStatusCode.NoContent)]
-        [ProducesResponseType((int) HttpStatusCode.BadRequest)]
+        [ProducesResponseType((int)HttpStatusCode.NoContent)]
+        [ProducesResponseType((int)HttpStatusCode.BadRequest)]
         public async Task<IActionResult> AddManualCorrectionToAccommodation(int accommodationId,
```
Comment (oleg1std) [inline HappyTravel.Nakijin.Api/Controllers/AccommodationManagementController.cs:61]: SetManualCorrectionStatus ?

File: HappyTravel.Nakijin.Api/Controllers/AccommodationManagementController.cs [hunk 2]
```diff
@@ -202,8 +203,28 @@ public async Task<IActionResult> DeactivateAccommodations([FromBody] Deactivatio
...
+        /// Removes dependency from supplier accommodation
+        /// </summary>
+        /// <param name="accommodationId"></param>
+        /// <param name="supplier"></param>
+        /// <param name="supplierCode"></param>
+        /// <returns></returns>
+        [HttpPost("accommodations/{accommodationId}/suppliers/{supplier}/accommodations/{supplierCode}/remove")]
+        [ProducesResponseType((int)HttpStatusCode.NoContent)]
+        [ProducesResponseType((int)HttpStatusCode.BadRequest)]
+        public async Task<IActionResult> RemoveSupplierAccommodation(int accommodationId, Suppliers supplier, string supplierCode)
```
Comment (oleg1std) [inline HappyTravel.Nakijin.Api/Controllers/AccommodationManagementController.cs:218]: I think the names in the controller could be simplified.
All the methods in the controller are about actions related to accommodation.

File: HappyTravel.Nakijin.Api/Services/AccommodationManagementService.cs [hunk 3]
```diff
@@ -25,12 +25,93 @@ namespace HappyTravel.Nakijin.Api.Services
...
             _accommodationChangePublisher = accommodationChangePublisher;
-            _logger = loggerFactory.CreateLogger<AccommodationManagementService>();
+            _logger = logger;
+        }
+
+
+        public async Task<Result> RemoveSupplierAccommodation(int accommodationId, Suppliers supplier, string supplierCode)
+        {
+            var utcDate = DateTime.UtcNow;
+            var accommodation = await _context.Accommodations.Where(a => a.Id == accommodationId && a.IsActive)
```
Comment (oleg1std) [inline HappyTravel.Nakijin.Api/Services/AccommodationManagementService.cs:40]: A predicate could be defined in the FirstOrDefaultAsync method

File: HappyTravel.Nakijin.Api/Services/AccommodationManagementService.cs [hunk 4]
```diff
@@ -25,12 +25,93 @@ namespace HappyTravel.Nakijin.Api.Services
...
+                    $"Accommodation with id {accommodationId} must have dependency from accommodation of supplier {supplier} with code {supplierCode}");
+
+            accommodation.SupplierAccommodationCodes.Remove(supplier);
+            accommodation.IsCalculated = false;
+            accommodation.Modified = utcDate;
+            _logger.LogRemovedDependencyFromSupplier(accommodationId, supplier.ToString(), supplierCode);
+
+            var htMapping = await _context.HtAccommodationMappings
+                .Where(m => m.IsActive && m.HtId == accommodationId)
+                .SingleOrDefaultAsync();
```
Comment (oleg1std) [inline HappyTravel.Nakijin.Api/Services/AccommodationManagementService.cs:60]: The "Where" method could be skipped. The predicate could be set in SingleOrDefaultAsync

File: HappyTravel.Nakijin.Api/Controllers/AccommodationManagementController.cs (hunk 1 above)
Comment (hasmikharutyunyan) [reply HappyTravel.Nakijin.Api/Controllers/AccommodationManagementController.cs:61]: It is not related with this PR
This api is for set manual static data for accommodation.

Comment (hasmikharutyunyan) [reply HappyTravel.Nakijin.Api/Controllers/AccommodationManagementController.cs:61]: Maybe when this will be used on admin panel, we can rename

**Review by oleg1std — APPROVED** [review]

**Pull request merged**